climada/hazard/tc_surge_bathtub.py

```diff
@@ -0,0 +1,179 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define TCSurgeBathtub class.
+"""
+
+__all__ = ['TCSurgeBathtub']
+
+import copy
+import logging
+
+import numpy as np
+import rasterio.warp
+import scipy.sparse as sp
+
+from climada.hazard.base import Hazard
+import climada.util.coordinates as coord_util
+
+LOGGER = logging.getLogger(__name__)
+
+HAZ_TYPE = 'TCSurgeBathtub'
+"""Hazard type acronym for this module"""
+
+MAX_DIST_COAST = 50
+"""Maximum inland distance of the centroids in km"""
+
+MAX_ELEVATION = 10
+"""Maximum elevation of the centroids in m"""
+
+DECAY_RATE_M_KM = 0.2
+"""Decay rate of surge in meters per each km"""
+
```

Comment:
Well spotted. I decided to have the decay rate as a user parameter in my latest commit.

However, I think that `MAX_DIST_COAST` and `MAX_ELEVATION` shouldn't be considered modeling choices. They are chosen so high, that they don't interfere with the model mechanics. Their purpose is to exclude non-relevant centroids early-on for performance reasons. Independent of any surge model, there should never be any surge for centroids lying that high or that far from the coast. Including them as keyword arguments doesn't improve usability. We have to remember that too many function arguments can overwhelm and confuse users.

Reply:
Does this also apply to very flat places like e.g. the Netherlands?

Reply:
Yes, because of the decay. Even with a low decay rate and a very low-lying point (5-6 meters below sea level), the surge will be gone after 50 km. However, the Netherlands are never affected by tropical cyclones, so it doesn't really make sense to go into the details.